saleor/payment/utils.py

```diff
@@ -67,6 +69,33 @@
 GENERIC_TRANSACTION_ERROR = "Transaction was unsuccessful"
 ALLOWED_GATEWAY_KINDS = {choices[0] for choices in TransactionKind.CHOICES}
 
+ALLOWED_MISSING_PSP_REFERENCE_MAP = {
+    WebhookEventSyncType.TRANSACTION_REFUND_REQUESTED: [
+        TransactionEventType.REFUND_FAILURE
+    ],
+    WebhookEventSyncType.TRANSACTION_CHARGE_REQUESTED: [
+        TransactionEventType.CHARGE_FAILURE
+    ],
+    WebhookEventSyncType.TRANSACTION_CANCELATION_REQUESTED: [
+        TransactionEventType.CANCEL_FAILURE
+    ],
+    WebhookEventSyncType.TRANSACTION_INITIALIZE_SESSION: [
+        TransactionEventType.CHARGE_REQUEST,
+        TransactionEventType.AUTHORIZATION_REQUEST,
+        TransactionEventType.AUTHORIZATION_ACTION_REQUIRED,
+        TransactionEventType.CHARGE_FAILURE,
+        TransactionEventType.AUTHORIZATION_FAILURE,
+    ],
```

Comment:
This list is missing `TransactionEventType.CHARGE_ACTION_REQUIRED` (it's in process_session):

```suggestion
    WebhookEventSyncType.TRANSACTION_INITIALIZE_SESSION: [
        TransactionEventType.CHARGE_REQUEST,
        TransactionEventType.AUTHORIZATION_REQUEST,
        TransactionEventType.CHARGE_ACTION_REQUIRED,
        TransactionEventType.AUTHORIZATION_ACTION_REQUIRED,
        TransactionEventType.CHARGE_FAILURE,
        TransactionEventType.AUTHORIZATION_FAILURE,
    ],
```